### PR TITLE
[INLONG-5312][Sort] Add unified inlong.metric parameter injection support for nodes

### LINKED ISSUE #5312: [Feature][Sort] Add unified inlong.metric parameter injection support for nodes
### Description

At present, many connectors support indicator reporting, but we have not added relevant metric reporting parameters at the entrance. And so, I add unified inlong.metric parameter injection support for nodes.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yunqingmoswu**
### Prepare a Pull Request
*(Change the title refer to the following example)*

Title: [INLONG-5312][Sort] Add unified inlong.metric parameter injection support for nodes

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

Fixes #5312 

### Motivation

At present, many connectors support indicator reporting, but we have not added relevant metric reporting parameters at the entrance. And so, I add unified `inlong.metric` parameter injection support for nodes.

### Modifications

The metric parameter `inlong.metric` will be added to the node options auto if it has implemented the interface InLongMetric.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Comment by gong** [comment]
PostgresExtractNode and HBaseLoadNode modify connector option

**Comment by yunqingmoswu** [comment]
> PostgresExtractNode and HBaseLoadNode modify connector option

It is better to mention PR separately later.

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/PostgresExtractNode.java [hunk 1]
```diff
@@ -100,7 +101,7 @@ public PostgresExtractNode(@JsonProperty("id") String id,
      *
      * @return options
      * @see <a href="https://ververica.github.io/flink-cdc-connectors/master/content/connectors/postgres-cdc.html">postgres
-     * cdc</a>
+     *         cdc</a>
```
Comment (EMsnap) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/PostgresExtractNode.java:104]: extra blank

File: inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java [hunk 2]
```diff
@@ -159,6 +162,23 @@ private void parseStream(StreamInfo streamInfo) {
...
+     * Inject the `inlong.metric` for ExtractNode or LoadNode
+     *
+     * @param streamInfo The encapsulation of nodes and node relations
+     */
+    private void injectInLongMetric(StreamInfo streamInfo) {
+        streamInfo.getNodes().stream().filter(node -> node instanceof InLongMetric).forEach(node -> {
+            Map<String, String> properties = node.getProperties();
+            if (properties == null) {
+                properties = new LinkedHashMap<>();
+            }
```
Comment (gong) [inline inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java:175]: if properties is null, new LinkedHashMap should be writed back to node by node.setProperties

**Review by thesumery — APPROVED** [review]

File: inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java (hunk 2 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java]: done

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/PostgresExtractNode.java (hunk 1 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/PostgresExtractNode.java]: done

**Review by EMsnap — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**